src/app/views/river-index/components/nwi-map.vue

```diff
@@ -480,7 +522,9 @@ export default {
     }
   },
   mounted () {
-    this.mountMap()
+    if (!this.isLoadingTileserverToken) {
```

Comment:
so I think this works and shouldn't have any holes in it (scenarios where the map initializes without the token when it should have the token), but I wonder if it'd make more sense to leverage Promises or async/await API here instead. Then you wouldn't need to store an "is loading" boolean either. Consider this:

```
async mounted () {
  if (this.isAmplifyDeploy) {
    await this.getTileserverToken(); // define this method with lines 491-504 from above
  }
  this.mountMap();
}
```

You could also use the Promise API (make `getTileserverToken` return a promise, then chain `getTileserverToken().then(response => { do the map mounting with your token });`, but I think async/await is cleaner.

Thoughts?